Extensions/XMPPMUCSUB/XMPPMUCSUB.m

```diff
@@ -750,36 +751,37 @@ - (void)xmppStream:(XMPPStream *)sender didReceiveMessage:(XMPPMessage *)message
      */
     
     NSXMLElement *event = [message elementForName:kXMPPEvent xmlns:XMPPPubSubEventNamespace];
-    NSXMLElement *messageItems = [event elementForName:kXMPPItems xmlns:XMPPMucSubMessageNamespace];
-    NSXMLElement *presenceItems = [event elementForName:kXMPPItems xmlns:XMPPMucSubPresenceNamespace];
-    NSXMLElement *subscribeItems = [event elementForName:kXMPPItems xmlns:XMPPMucSubSubscribeNamespace];
-    NSXMLElement *unsubscribeItems = [event elementForName:kXMPPItems xmlns:XMPPMucSubUnsubscribeNamespace];
-    
-    if (messageItems) {
-        NSXMLElement *item = [messageItems elementForName:kXMPPItem];
-        NSXMLElement *messageElement = [item elementForName:kXMPPMessage];
-        if (messageElement) {
-            XMPPMessage *mucSubMessage = [XMPPMessage messageFromElement:messageElement];
-            [multicastDelegate xmppMUCSUB:self didReceiveMessage:mucSubMessage];
+    NSXMLElement *items = [event elementForName:kXMPPItems];
+    
+    if (items) {
+        NSXMLElement *item = [items elementForName:kXMPPItem];
+        NSString *node = [[items attributeForName:kXMPPNode] stringValue];
+        if ([node isEqualToString:XMPPMucSubMessageNamespace]) {
+            NSXMLElement *messageElement = [item elementForName:kXMPPMessage];
+            if (messageElement) {
+                XMPPMessage *mucSubMessage = [XMPPMessage messageFromElement:messageElement];
+                [multicastDelegate xmppMUCSUB:self didReceiveMessage:mucSubMessage];
+            }
         }
-    }
-    if (presenceItems) {
-        NSXMLElement *item = [presenceItems elementForName:kXMPPItem];
-        NSXMLElement *presenceElement = [item elementForName:kXMPPPresence];
-        if (presenceElement) {
-            XMPPPresence *mucSubPresence = [XMPPPresence presenceFromElement:presenceElement];
-            [multicastDelegate xmppMUCSUB:self didReceivePresence:mucSubPresence];
+        else if ([node isEqualToString:XMPPMucSubPresenceNamespace]) {
+            NSXMLElement *presenceElement = [item elementForName:kXMPPPresence];
+            if (presenceElement) {
+                XMPPPresence *mucSubPresence = [XMPPPresence presenceFromElement:presenceElement];
+                [multicastDelegate xmppMUCSUB:self didReceivePresence:mucSubPresence];
+            }
+        }
+        else if ([node isEqualToString:XMPPMucSubSubscribeNamespace]) {
+            NSXMLElement *subscribeElement = [item elementForName:kXMPPSubscribe];
+            if (subscribeElement) {
+                [multicastDelegate xmppMUCSUB:self didSubscribe:subscribeElement];
+            }
+        }
+        else if ([node isEqualToString:XMPPMucSubUnsubscribeNamespace]) {
+            NSXMLElement *unsubscribeElement = [item elementForName:kXMPPUnsubscribe];
+            if (unsubscribeElement) {
+                [multicastDelegate xmppMUCSUB:self didUnsubscribe:unsubscribeElement];
+            }
         }
```

Comment:
Looks good to me now. Should we have an `else` that handles unexpected elements?  Maybe just prints a log message or something?

Reply:
Hmm.. muc-sub is only handling these four types but there could be other types specifically for other modules. Adding a log here would log a message for every other types that are being handled by other modules and not by muc-sub. I think it could be confusing. I was thinking to log something like: _This item node is not being handled by muc-sub:_ but still this will show up each time pubsub is sending another type of message for other modules.. Any thoughts?!!